### PR TITLE
[FLINK-3160] [web-dashboard] Aggregate operator statistics by TaskManager

### PR DESCRIPTION
**Pull request by greghogan**
Adds a new per-job tab displaying task statistics aggregated by TaskManager

### CODE REVIEW

**Comment by StephanEwen** [comment]
Could you describe what exactly the change looks like, share a few screenshots of the parts that changed?

**Comment by greghogan** [comment]
Running WordCount on a single node running two taskmanagers, the first screenshot is the current UI (with 'Overview' renamed to 'Subtasks') and the second screenshot is the new tab for 'TaskManagers'.

Hostname is appended with port in both tabs in order to disambiguate TaskManagers, although this deserves further consideration. It would be cleaner to use a per-host index for TaskManagers rather than the 5-digit port (ie., 127.0.0.1:0, 127.0.0.1:1, ...).

Subtasks and TaskManagers are now sorted by host.

Subtasks

TaskManagers

**Comment by fhueske** [comment]
I did not look at the code, but the screenshots look very good.
+1 for renaming "Overview" to "Subtasks" and adding the TaskManager tab. 
I'm OK with using ports for now.

File: flink-runtime-web/web-dashboard/app/partials/jobs/job.plan.jade [hunk 1]
```diff
@@ -22,7 +22,10 @@
   nav.navbar.navbar-default.navbar-secondary-additional
     ul.nav.nav-tabs
       li(ui-sref-active='active')
-        a(ui-sref=".overview({nodeid: nodeid})") Overview
+        a(ui-sref=".overview({nodeid: nodeid})") Subtasks
```
Comment (uce) [inline flink-runtime-web/web-dashboard/app/partials/jobs/job.plan.jade]: Should we rename the `overview` ref to `subtasks` as well?

File: flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java [hunk 2]
```diff
@@ -203,6 +204,27 @@ public JobVertexID getJobVertexId() {
 	public ExecutionVertex[] getTaskVertices() {
 		return taskVertices;
 	}
+
+	public Map<String, List<ExecutionVertex>> getTaskVerticesByTaskManager() {
```
Comment (uce) [inline flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java]: Can we move this to the handler? I think that will be the only place where we use it in the future as well.

**Comment by uce** [comment]
Changes and screenshots look very good!

I have a minor inline comment regarding the added method. After addressing it, I think that this is good to merge.

**Comment by greghogan** [comment]
@uce I rebased, made the noted changes, and updated the pull request.

**Comment by uce** [comment]
Thank you Greg! Let me try this out once more and then merge it if everything works as expected.

**Pull request closed without merging**